src/core/funcbloc.cpp

```diff
@@ -671,6 +674,18 @@ size_t CFunctionBlock::getToStringBufferSize() const {
 
 }
 
+void CFunctionBlock::triggerEventsOfType(TEventTypeID paEventTypeId) {
+  //most of the FBs will only have the basic event type -> mEITypes == nullptr
+  if (getFBInterfaceSpec().mEITypeNames != nullptr) {
+    for (TEventID eventId = 0; eventId < getFBInterfaceSpec().mNumEIs; eventId++) {
+      if (getEIType(eventId) == paEventTypeId && !isInputEventConnected(eventId)) {
+        getResource()->getResourceEventExecution()->startEventChain(CConnectionPoint(this, eventId));
```

Comment:
use the eventchain execution thread which was used to triggere the R_TRIG fb and add the events to the internal event list.